src/app/(proper_react)/(redesign)/(authenticated)/user/(dashboard)/dashboard/fix/data-broker-profiles/removal-under-maintenance/RemovalUnderMaintenanceView.tsx

```diff
@@ -0,0 +1,397 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+"use client";
+
+import {
+  StepDeterminationData,
+  getNextGuidedStep,
+} from "../../../../../../../../../functions/server/getRelevantGuidedSteps";
+import { FixView } from "../../FixView";
+import styles from "./RemovalUnderMaintenance.module.scss";
+import { useL10n } from "../../../../../../../../../hooks/l10n";
+import { TelemetryLink } from "../../../../../../../../../components/client/TelemetryLink";
+import { TelemetryButton } from "../../../../../../../../../components/client/TelemetryButton";
+import { CONST_URL_SUMO_MANUAL_REMOVAL } from "../../../../../../../../../../constants";
+import React, { useEffect, useState } from "react";
+import {
+  BackArrow,
+  ClockIcon,
+} from "../../../../../../../../../components/server/Icons";
+import { LatestOnerepScanData } from "../../../../../../../../../../db/tables/onerep_scans";
+import { ExposureCardDataClassLayout } from "../../../../../../../../../components/client/exposure_card/ExposureCardDataClass";
+import confetti from "canvas-confetti";
+import { OnerepScanResultRow } from "knex/types/tables";
+import { toast } from "react-toastify";
+
+export type Props = {
+  data: LatestOnerepScanData;
+  stepDeterminationData: StepDeterminationData;
+  subscriberEmails: string[];
+};
+
+export const RemovalUnderMaintenanceView = (props: Props) => {
+  const l10n = useL10n();
+  const [detailedRemovalGuide, setDetailedRemovalGuide] = useState(false);
+  const [firstScanResultNotResolved, setFirstScanResultNotResolved] = useState(
+    props.data.results[0],
+  );
+  const [isLoadingNextDataBroker, setIsLoadingNextDataBroker] = useState(false);
+  const [fadeState, setFadeState] = useState<"fadeIn" | "fadeOut" | "">("");
+
+  const nextGuidedStep = getNextGuidedStep(
+    props.stepDeterminationData,
+    "DataBrokerManualRemoval",
+  );
+
+  const [nextScanResultNotResolved, setNextScanResultNotResolved] = useState<
+    OnerepScanResultRow | undefined
+  >();
+
+  const updateDataBrokerWithTimeout = () => {
+    setIsLoadingNextDataBroker(true);
+    setFadeState("fadeOut");
+
+    setTimeout(() => {
+      setIsLoadingNextDataBroker(false);
+      void confetti();
+      setFadeState("fadeIn"); // trigger the fade-in for the next scan result
+    }, 2000);
+  };
+
+  async function handleManualRemovalChange() {
```

Comment:
It is safer to handle potential errors here when calling `fetch`. If this function throws an error we can’t recover. Let’s wrap the body of `handleManualRemovalChange` in a `try...catch`.